### PR TITLE
Dockerfile + build process + daemonset + README updates

### LINKED ISSUE #1: Add Dockerfile
The device plugin should be running in a container as a daemonset.

### LINKED ISSUE #6: README file!
[stub]

### PR DESCRIPTION
**Pull request by dougbtv**
Adds:

* Updates to readme, showing some sample usage, sample build
* Daemonset to build the pod
* Dockerfile to build an image
* Travis for lint & also to push docker

Some of these things aren't properly namespaced -- due to the CI side of things, and refer to kind of "doug & nfvpe" so, we need to move this to the NFVPE repo to fix those up. I created #7 to follow up on this. However, should work with the Daemonset as advertised in the readme, had a test through that.

Fixes #1
Fixes #6

### CODE REVIEW

**Comment by dougbtv** [comment]
Just an FYI that because of some wonkiness in how the secrets work, this will have to be merged, and then I'll update the secrets.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# deploy:
+#   provider: script
+#   script: docker push nfvpe/dummy-dp
+#   on:
+#     branch: master
+
+# Using this for now so it triggers every time other tests succeed.
+after_success:
+  - docker login -u "$REGISTRY_USER" -p "$REGISTRY_PASS"
+  - docker build -t nfvpe/dummy-dp .
```
Comment (fepan) [inline .travis.yml]: I'd prefer a name like 'dummy-device-plugin' if that's possible

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+before_script:
+  - golint .
+  - go fmt .
+  - go vet .
+  # was failing with: `gocyclo: vendor/golang.org/x/net/context/go19.go:15:14: expected type, found '=' (and 1 more errors)`
+  # - gocyclo -over 15 .
+
+script:
+  - go build dummy.go
```
Comment (fepan) [inline .travis.yml]: just do go build here and generate a binary called k8s-dummy-device-plugin. Same for other places where we build and run binary.

File: .travis.yml (hunk 1 above)
Comment (dougbtv) [reply .travis.yml]: sure thing, have it updated to be `nfvpe/dummy-device-plugin`

File: .travis.yml (hunk 2 above)
Comment (dougbtv) [reply .travis.yml]: cool, updated the name of the resulting binary with the `-o output_binary_name_here` option for `go build`

**Pull request merged**